### PR TITLE
Buildfix OpenBSD.

### PR DESCRIPTION
**Pull request by 3405691582**
Of note, the config header is just forked from the Linux config file and
mutated until things build correctly. I didn't think it necessary to
spend too much effort to make sure the config header matches the output
from configure too much.

### CODE REVIEW

**Comment by jmillikin** [comment]
Thanks for the patch. I made two small tweaks -- could you review them at your convenience?
1. The original `config-openbsd.h` produced some build warnings about undeclared symbols, so I adjusted it based on the output of `./configure`. 
2. The `ctx.os` value in a repository rule is set based on the build host, not the target. To enable cross-compilation, target-specific files such as the stubbed `alloca.h` should be handled via `select()`.

Also, how would you like to be credited in Git history + AUTHORS? Your branch is currently credited to your GitHub username, but I could adjust it to match OpenBSD community participation instead if you'd prefer (for example https://marc.info/?l=openbsd-tech&m=160138988317298&w=2).

(I'll apply the same answers to https://github.com/jmillikin/rules_bison/pull/11)

File: m4/internal/gnulib/config-openbsd.h [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+#define HAVE_READLINK 1
+#define HAVE_REALPATH 1
+#define HAVE_SECURE_GETENV 1
+#define HAVE_SETRLIMIT 1
+#define HAVE_SIGACTION 1
+#define HAVE_SIGALTSTACK 1
+#define HAVE_SIGINTERRUPT 1
+#define HAVE_SIGSET_T 1
+#define HAVE_SIG_ATOMIC_T 1
+#define HAVE_SNPRINTF 1
```
Comment (3405691582) [inline m4/internal/gnulib/config-openbsd.h]: Setting this back to 0 will avoid a crash on this platform (see https://github.com/openbsd/ports/blob/master/devel/m4/patches/patch-lib_vasnprintf_c which handles this differently, but implementing this same particular patch conditionally like this probably requires more work that I'm not sure is necessary right now). This needs to be done on bison too.

**Comment by 3405691582** [comment]
Other than the HAVE_SNPRINTF comment, the rest LGTM. 

I also don't mind if you use my Github username.

**Comment by jmillikin** [comment]
This patch has been included in release v0.2.2: https://github.com/jmillikin/rules_m4/releases/tag/v0.2.2

**Pull request closed without merging**